src/Compilers/Core/Portable/SourceGeneration/Nodes/SyntaxValueProvider_ForAttributeWithSimpleName.cs

```diff
@@ -39,7 +39,11 @@ public partial struct SyntaxValueProvider
     /// <c>context.SyntaxProvider.CreateSyntaxProviderForAttribute(nameof(CLSCompliantAttribute), (node, c) => node is ClassDeclarationSyntax)</c>
     /// will find the <c>C</c> class.
     /// </summary>
-    internal IncrementalValuesProvider<SyntaxNode> ForAttributeWithSimpleName(
+    /// <remarks>
+    /// Note: a 'Values'-provider of arrays are returned.  Each array provides all the matching nodes from a single <see
+    /// cref="SyntaxTree"/>.
+    /// </remarks>
+    internal IncrementalValuesProvider<ImmutableArray<SyntaxNode>> ForAttributeWithSimpleName(
```

Comment:
note that instead of `IncrementalValuesProvider<SyntaxNode>` which represents *all* matching nodes, we get a Value-provider of `ImmutableArray<SyntaxNode>` where each `ImmutableArray` corresponds to the matching nodes from *one* SyntaxTree.  This reformalization simplifies things both in this file and in the consumption side in 'ForAttributeWithMetadataName'.